README.md

```diff
@@ -0,0 +1,560 @@
+
+NormalizedCacheFactory<LruNormalizedCache> cacheFactory = new LruNormalizedCacheFactory(EvictionPolicy.NO_EVICTION)
+        .chain(new SqlNormalizedCacheFactory(apolloSqlHelper));
+ ```
+ - Passing in connection params for subscriptions to work if server is behind authentication mechanism
+ 
+ ```java
+ Map<String, Object> connectionParams = new HashMap<>();
+connectionParams.put("Authorization", authHeader);
+```
+- And build our Apollo Client
+
+```java
+return ApolloClient.builder()
+        .enableAutoPersistedQueries(true)
+        .serverUrl(serverUrl)
+        .normalizedCache(cacheFactory, resolver)
+        .subscriptionConnectionParams(connectionParams)
+        .subscriptionTransportFactory(new WebSocketSubscriptionTransport.Factory(serverUrl, okHttpClient))
+        .okHttpClient(okHttpClient)
+        .build();
+```
+
+#### Using queries, mutation and subscriptions
+Once client is build we can use it to run queries, mutations and subscriptions. On application start, a query, mutation or subscription are build
+#### Query
+
+```java
+AllTasksQuery tasksQuery = AllTasksQuery
+        .builder()
+        .build();
+  ```
+#### Mutation
+
+ ```java
+DeleteTaskMutation deleteTask = DeleteTaskMutation
+        .builder()
+        .id(Param that is passed in to identify item to remove)
+        .build();
+ ```
+
+#### Subscriptions
+```java
+DeleteTaskSubscription deleteTaskSubscription = DeleteTaskSubscription
+        .builder()
+        .build();
+   ```
+
+   Next, use created client to initialize built query, mutation, or subscription and send request to the server.
+#### Query
+  ```java
+  client.query(tasksQuery)
+        .responseFetcher(onlineResponse)
+        .enqueue(new ApolloCall.Callback<MyGeneratedQuery.Data>()
+  ```
+- `.responseFetcher` in our example has either `NETWORK_ONLY` or `CACHE_ONLY` policy. Depending on device being online or offline. If `online` all data used in query is coming from the server, if `offline` all data is coming from normalized cache.
+#### Mutation
+ ```java
+ client.mutate(deleteTask)
+        .refetchQueries(tasksQuery)
+        .enqueue(new ApolloCall.Callback<DeleteTaskMutation.Data>() {
+```
+- We want to use `refetchQueries(myGetQuery)` in mutations just to refetch our get query so cache gets updated.
+
+#### Subscriptions
+   ```java
+   client.subscribe(deleteTaskSubscription)
+        .execute(new ApolloSubscriptionCall.Callback<DeleteTaskSubscription.Data>() {
+  ```
+- Currently the only way to refresh cache after receiving subscription response is to use `subscribeToMore`, however, this is not available in java.
+  After request has been sent we either receive `onResponse`, if request was successful, or `onFailure` if request has failed.
+
+ ```java
+  @Override
+public void onResponse(@NotNull Response<AllTasksQuery.Data> response) {
+    ON RESPONSE LOGIC
+}
+ @Override
+public void onFailure(@NotNull ApolloException e) {
+    ON FAILURE LOGIC
+}
+```
```

Comment:
is there an Apollo android docs that we can link here so that if people wants to know more they will be able to find out themselves.